tests/unit/test_diffsync.py

```diff
@@ -468,6 +468,26 @@ def callback(stage, current, total):
     assert last_value == {"current": expected, "total": expected}
 
 
+def test_diffsync_sync_to_w_diff(backend_a, backend_b):
+    diff = backend_b.diff_to(backend_a)
+    assert diff.has_diffs()
+    # Mock diff_from to make sure it's not called when passing in an existing diff
+    backend_b.diff_from = mock.Mock()
+    # Perform full sync
+    backend_b.sync_to(backend_a, diff=diff)
+    assert not backend_b.diff_from.called
```

Comment:
Might also be good to assert that the sync actually updates `backend_a` according to the provided diff?

Reply:
I had to create this within a new test since we're mocking within this test, but it's there now to do post diff and assert it doesn't have any diffs.